modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/Select/Select.es.js

```diff
@@ -218,6 +223,61 @@ const DropdownItem = ({
 	</>
 );
 
+const DropdownWithSearch = ({
+	currentValue,
+	expand,
+	handleSelect,
+	multiple,
+	options,
+}) => {
+	const [query, setQuery] = useState('');
+	const [filteredItems, setFilteredItems] = useState([]);
+
+	const emptyOption = {
+		label: Liferay.Language.get('choose-an-option'),
+		value: null,
+	};
+
+	useEffect(() => {
+		const result = options.filter(
+			(option) =>
+				option.value &&
+				option.label.toLowerCase().includes(query.toLowerCase())
+		);
+		setFilteredItems([emptyOption, ...result]);
+		// eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
Yes. I saw in other examples that Matu puts this line to avoid eslint adding more parameters beside this 'query' parameter, which was making the code get in infinity loop.